Frameworks/Foundation/NSUndoManager.mm

```diff
@@ -38,11 +91,19 @@ - (id)prepareWithInvocationTarget:(id)target {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (void)undo {
-    UNIMPLEMENTED();
+    if (![self canUndo]) {
+        return;
+    }
+    _undoing = YES;
+
+    _NSUndoGroup* undoGroup = (_NSUndoGroup*)[_undoStack lastObject];
+    [undoGroup callUndoGroup];
```

Comment:
The cast should be unnecessary with generic syntax. If it's necessary, please file a bug to get NSArray fixed.